extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeSet.php

```diff
@@ -1,24 +1,40 @@
 <?php
 namespace Wikia\PortableInfobox\Parser\Nodes;
 
-use Wikia\PortableInfobox\Parser\XmlParser;
-
 class NodeSet extends Node {
 
 	public function getData() {
-		$nodeFactory = new XmlParser( $this->infoboxData );
-		if ( $this->externalParser ) {
-			$nodeFactory->setExternalParser( $this->externalParser );
+		if ( !isset( $this->data ) ) {
+			$this->data = [ 'value' => $this->getDataForChildren() ];
 		}
-		return [ 'value' => $nodeFactory->getDataFromNodes( $this->xmlNode, $this ) ];
+
+		return $this->data;
+	}
+
+	public function getRenderData() {
+		return [
+			'type' => $this->getType(),
+			'data' => [
+				'value' => array_map( function ( Node $item ) {
```

Comment:
Here I think that putting each argument of array_map in its separate line will help in reading this code.